R/inspectDifferences.R

```diff
@@ -25,43 +28,51 @@
 #' equalGADS(pisa, pisa2)
 #'
 #' # inspect via inspectDifferences()
-#' inspectDifferences("age", GADSdat1 = pisa, GADSdat2 = pisa2, id = "idstud")
+#' inspectDifferences(GADSdat = pisa, varName = "age", other_GADSdat = pisa2, id = "idstud")
 #'
 #'@export
-inspectDifferences <- function(varName, GADSdat1, GADSdat2, id) {
-  check_GADSdat(GADSdat1)
-  check_GADSdat(GADSdat2)
-  if(!is.character(varName) || length(varName) != 1) stop("'varName' must be a character of length 1.")
-  if(!is.character(id) || length(id) != 1) stop("'id' must be a character of length 1.")
-  if(!varName %in% namesGADS(GADSdat1)) stop("'varName' is not a variable in 'GADSdat1'.")
-  if(!varName %in% namesGADS(GADSdat2)) stop("'varName' is not a variable in 'GADSdat2'.")
-  if(!id %in% namesGADS(GADSdat1)) stop("'id' is not a variable in 'GADSdat1'.")
-  if(!id %in% namesGADS(GADSdat2)) stop("'id' is not a variable in 'GADSdat2'.")
-  if(nrow(GADSdat1$dat) != nrow(GADSdat2$dat)) stop("'GADSdat1' and 'GADSdat2' have different row numbers.")
-  if(any(is.na(GADSdat1$dat[, id]))) stop("Missing values in 'id' column of 'GADSdat1'.")
-  if(any(is.na(GADSdat2$dat[, id]))) stop("Missing values in 'id' column of 'GADSdat2'.")
-  if(any(GADSdat1$dat[, id] != GADSdat2$dat[, id])) stop("'id' column is not equal for 'GADSdat1' and 'GADSdat2'.")
+inspectDifferences <- function(GADSdat, varName, other_GADSdat = GADSdat, other_varName = varName, id) {

```

Comment:
Hmm, just a thought. Are there cases, where something like "other_id" could be useful?

Reply:
Good point have not thought about this before! While I can imagine that there hypothetically might be some cases in which the id variables have different names, this IMHO would suggest that those are different id variables. For example, if the function is used to compare two `GADSdat` objects that have been prepared by two different persons, the variable naming should still be identical.

So will leave-as-is for now but I am happy to come back to this, if a specific use case comes up!